api/control/v1alpha1/types.go

```diff
@@ -302,6 +315,211 @@ type BindingList struct {
+type Expression struct {
+	Op   ExpressionOperator `json:"op"`
+	Path string             `json:"path,omitempty"`
+	Args []Expression       `json:"args,omitempty"`
+}
+
+type ExpressionOperator string
+
+const (
+	OperatorPath  ExpressionOperator = "Path" // JSONPath string
+	OperatorOr    ExpressionOperator = "Or"
+	OperatorAnd   ExpressionOperator = "And"
+	OperatorNot   ExpressionOperator = "Not"
+	OperatorEqual ExpressionOperator = "Equal"
+)
+
+// StatusCollectorStatus defines the observed state of StatusCollector.
+type StatusCollectorStatus struct {
+	ObservedGeneration int64 `json:"observedGeneration"`
+
+	// +optional
+	Errors []string `json:"errors,omitempty"`
+}
+
+// StatusCollectorList is the API type for a list of StatusCollector.
+//
+// +kubebuilder:object:root=true
+type StatusCollectorList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []StatusCollector `json:"items"`
+}
+
+// CombinedStatus holds the combined status from the WECs for one particular (workload object, BindingPolicy) pair.
+// The namespace of the CombinedStatus object is the namespace of the workload object,
+// or "kubestellar-report" if the workload object has no namespace.
+// The name of the CombinedStatus object is the concatenation of:
+// - the NamespacedName the workload object
+// - the string ":"
+// - the Name of the BindingPolicy object.
+// The CombinedStatus object has the following labels:
+// - "status.kubestellar.io/api-group" holding the API Group (not verison) of the workload object;
+// - "status.kubestellar.io/resource" holding the resource (lowercase plural) of the workload object;
+// - "status.kubestellar.io/namespace" holding the namespace of the workload object;
+// - "status.kubestellar.io/name" holding the name of the workload object;
+// - "status.kubestellar.io/binding-policy" holding the name of the BindingPolicy object.
+//
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:shortName={cs}
+// +kubebuilder:printcolumn:name="GROUP",type="string",JSONPath=".metadata.labels['status\\.kubestellar\\.io/api-group']"
+// +kubebuilder:printcolumn:name="RSC",type="string",JSONPath=".metadata.labels['status\\.kubestellar\\.io/resource']"
+// +kubebuilder:printcolumn:name="NS",type="string",JSONPath=".metadata.labels['status\\.kubestellar\\.io/namespace']"
+// +kubebuilder:printcolumn:name="NAME",type="string",JSONPath=".metadata.labels['status\\.kubestellar\\.io/name']"
+// +kubebuilder:printcolumn:name="BINDINGPOLICY",type="string",JSONPath=".metadata.labels['status\\.kubestellar\\.io/policy']"
+type CombinedStatus struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// `results` has an entry for every applicable NamedStatusCombiner.
```

Comment:
This text needs updating, "NamedStatusCombiner" is not a thing in this PR.

Reply:
done.